### DISCUSSION TITLE
[Unstable] Bots attempt to use wreck/enemy sub deconstructors if the deconstructor is inaccessible

### DISCUSSION THREAD

**Discussion by HawkinsThe71st** [FakeFishGames/Barotrauma#13607] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

If the deconstructor is inaccessible, either because it is already in use or set to be ignored, bots will say they can't reach any deconstructors that are outside the sub. I assume this is to enable them to use outpost deconstructors but on regular levels with wrecks or hostile subs they will periodically say they cannot reach their deconstructors.
Level with a hostile sub:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/fe7e5f31-4f7a-404d-a4f6-cd78d6149dd1)
Level with a wrecked sub:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/669da249-cc47-41a7-81b0-44a25ce4a91a)

A suggestion for the deconstruct order:
I think when marking items for deconstruction it should mark the entire stack of items instead of just a single one. Otherwise when you order a bot to deconstruct 8 pieces of ore, you have to mark each one for deconstruction at which point it is quicker and less tedious to go put the ore into the deconstructor manually.

### Reproduction steps

1. Order a bot to deconstruct
2. Make the sub deconstructor inaccessible to bots either by filling it with other items or applying the "ignore this" order
3. Mark an item for deconstruction for the bot

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.13.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I'm not entirely sure if this is actually a bug. The bots intentionally say this line when there's no deconstructor available in the sub or a connected friendly outpost, it doesn't make a difference whether there are deconstructors in wrecks or enemy subs.

Giving this a test, it seems to work correctly: the bots don't say the line if there's a usable deconstructor in the sub (this includes deconstructors that are already full of items).
